flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -586,7 +588,18 @@ private boolean performCheckpoint(CheckpointMetaData checkpointMetaData) throws
 				// yet be created
 				final CancelCheckpointMarker message = new CancelCheckpointMarker(checkpointMetaData.getCheckpointId());
 				for (ResultPartitionWriter output : getEnvironment().getAllWriters()) {
-					output.writeEventToAllChannels(message);
+					final Buffer eventBuffer = EventSerializer.toBuffer(message);
```

Comment:
in any case the entire try/finally block could be moved into the ResultPartitionWriter, correct?